glib/src/main_context_channel.rs

```diff
@@ -539,6 +539,7 @@ impl MainContext {
     /// will fail.
     ///
     /// The returned `Sender` behaves the same as `std::sync::mpsc::Sender`.
+    #[deprecated = "Use an async channel on the main context instead"]
```

Comment:
If there is a name for the function/method, that'd be quite helpful I think.

Reply:
There are like 5 different crates that people could use. Not sure we want to prescribe one here.

Reply:
Also, why do we want to force people to use async when this code works perfectly fine and is much better to use (imo)? I don't remember the discussion about this.

Reply:
Because it's not much better to use and leads to worse code in practice. Ask all the GNOME apps that moved away from the main context channel.

Callbacks alone are already worse than async Rust, but in the context of GTK it makes everything a lot more awkward. But by us providing this API, people use that first and then wonder why it's all so awful :)

Reply:
Intriguing but my opinion doesn't really matter in the end as I'm not that much of gtk-rs user...